python/python/bystro/supervised_ppca/tests/test_ppca_regularized.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from bystro.supervised_ppca.ppca_regularized import PPCARegularized
+
+
+def test_initialization():
+    """Test the initialization of the PPCARegularized class."""
+    pca_model = PPCARegularized(n_components=2)
+    assert (
+        pca_model.n_components == 2
+    ), "n_components should be initialized correctly."
+
+
+def test_fit():
+    """Test the fitting process of the PPCARegularized model."""
+    # Create a synthetic dataset
+    rng = np.random.default_rng(2021)
+    X = rng.normal(size=(100, 5))  # 100 samples and 5 features
+
+    pca_model = PPCARegularized(
+        n_components=2, regularization_options={"method": "NonLinearShrinkage"}
+    )
+    pca_model.fit(X)
```

Comment:
Something is going wrong here:

```python
In [3]: pca_model.fit(X)
/mnt/ssd1/bystro/python/python/bystro/supervised_ppca/ppca_regularized.py:75: RuntimeWarning: invalid value encountered in sqrt
  W = eigenvectors[:, :L] * np.sqrt(
Out[3]: <bystro.supervised_ppca.ppca_regularized.PPCARegularized at 0x7fb0cace1c90>
```

Reply:
Forgot to reorder the eigenvalues as well. Issue is now fixed.